### ISSUE TITLE
useTitle not updating in React Concurrent Mode

### ISSUE THREAD

**Issue by tazsingh** [0no-co/hoofd#90]
I have yet to create a focused reproduction of this issue, but I am noticing that with React's Concurrent Mode, code similar to the following will never produce the latter:

```tsx
const MyComponent = ({ title }) => {
  console.log('Title Value', title)

  useTitle(title || 'fallback')

  return null
}
```

I'm essentially seeing the following `console.log`s:

```
Title Value undefined
Title Value undefined
Title Value undefined
Title Value The Title I Want
```

However, the `<title>` tag continue to be `fallback` 🧐 

Happy to pair up with you on what I'm seeing!

**Comment by JoviDeCroock** [comment]
Hmm I assume this might be another `StrictMode` bug happening due to effects being executed twice, or does this happen in production builds as well? If possible would love to see a reproduction here, else I can also take a look at making a codesandbox later

**Comment by tazsingh** [comment]
Posting here for reference from our DMs, but I tried replicating what I believe our app and its libraries to be doing under the hood in this Codesandbox and was also unable to replicate the behaviour I'm noticing 🤷 https://codesandbox.io/s/musing-bartik-8o834g?file=/src/App.tsx

I'll try debugging further, in our app and otherwise. Thanks again!

**Comment by ridwanrsk** [comment]
Hello @JoviDeCroock! First of all thanks for this library!

I forked this sandbox [from this thread](https://github.com/0no-co/hoofd/issues/4#issuecomment-614085388). Here's the fork: https://codesandbox.io/s/hoofd-with-react-18-lg4pjv?file=/src/index.js

After doing these things:
- Replacing `hooked-head` with `hoofd`
- Updating Codesandbox dependencies

I was able to get it work. However, in the thread it uses `ReactDOM.render` where `ReactDOM` is imported from `react-dom`. In newer projects, Codesandbox uses `createRoot` from `react-dom/client`. If you modify the index.js to use `createRoot`, it stops updating.

![hooked-head](https://github.com/0no-co/hoofd/assets/137581026/84d5d7cb-51d0-436a-ab18-bfc945c3c1f2)
